lib/docker.js

```diff
@@ -243,7 +248,26 @@ function startContainer(args, hidePorts) {
 
 }
 
+function ensureDockerVersion() {
+    return promisedSpawn(['docker', '-v'], { capture: true })
+    .then(function(dockerVersion) {
+        if (!dockerVersion) {
+            console.error('Docker is not found.');
+            process.exit(1);
+        }
+
+        dockerVersion = /(\d+\.\d+\.\d+)/.exec(dockerVersion);
+        if (!dockerVersion[1]) {
+            console.error("Couldn't find docker version. Make sure docker is installed");
+            process.exit(1);
+        }
 
+        if (os.type() === 'Darwin' && semver.lt(dockerVersion[1], '1.12.0')) {
```

Comment:
Nice check! Could you add one for the non-Darwin case as well? We need docker >= 1.8.0